### DISCUSSION TITLE
STRIDE Threat Model for EdgeX v3

### DISCUSSION THREAD

**Discussion by nyameen** [edgexfoundry/community#185] [Q&A]
Hello EdgeX community,

I was reading through some of the [V3 Migration](https://github.com/edgexfoundry/edgex-docs/blob/minnesota/docs_src/security/V3Migration.md?plain=1#L31-L41) guides and saw a mention to a STRIDE threat model for EdgeX v3. When referring to the [report](https://docs.edgexfoundry.org/3.0/threat-models/stride-model/EdgeXFoundryThreatReportV2.2.htm), it shows a threat model for version 2.2.

Are there any plans to update the model to reflect version 3.0?

Thank you!

**Comment by bnevis-i** [comment]
> Are there any plans to update the model to reflect version 3.0?

No.  The STRIDE threat model was done as an outside-in view, and isn't sufficiently detailed to capture the internal architecture of EdgeX.  Some of the big changes in EdgeX 3.0, like changing to a different API gateway, or adding internal microservice authentication, wouldn't be reflected in the threat model due to that reason, and are essentially swapping out one component for another.
